tests/arp/translation/test_arp_tree.py

```diff
@@ -0,0 +1,44 @@
+from nagini_contracts.lock import Lock
+from nagini_contracts.contracts import *
+from nagini_contracts.obligations import Level, WaitLevel, MustTerminate
+from nagini_contracts.thread import Thread, getARP
+
+
+class Cell:
+    def __init__(self, val: int) -> None:
+        self.value = val
+        Ensures(Acc(self.value) and self.value == val)
+
+
+class TreeTask:
+
+    def __init__(self, c: Cell) -> None:
+        self.data = c
+        self.result = 0
+        Ensures(Acc(self.data) and self.data == c)
+        Ensures(Acc(self.result) and self.result == 0)
+
+    def run(self) -> None:
+        Requires(Rd(self.data) and Rd(self.data.value) and Acc(self.result))
+        Ensures(Rd(self.data) and Rd(self.data.value) and Acc(self.result))
+        subs = Sequence()  # type: Sequence[Thread]
+        tasks = Sequence()  # type: Sequence[TreeTask]
+
+        i = 0
+        while i < 10:
+            #:: ExpectedOutput(not.supported)
```

Comment:
Did you add this way of expecting UnsupportedExceptions, has this existed for longer and I just wasn't aware of it, or is this somehow a mistake? I don't know this test annotation.

Reply:
I did not add anything and as it turns out it also does not work... What is the preferred way to specify a NotSupported exception being thrown?

Reply:
There isn't any. I think at one point the intention was that a NotSupported exception means that we plan to support something in the future but don't at the moment, and tests should only check actually intended behavior. To be honest, I'd just say that permission expressions have to have a specific format, and change the code to raise an InvalidProgramException with the code unsupported.permission.format or something like that instead.